config.go

```diff
@@ -0,0 +1,224 @@
+// Copyright 2022 Juan Pablo Tosso and the OWASP Coraza contributors
+// SPDX-License-Identifier: Apache-2.0
+
+package coraza
+
+import (
+	"io/fs"
+
+	"github.com/corazawaf/coraza/v3/internal/corazawaf"
+	"github.com/corazawaf/coraza/v3/loggers"
+	"github.com/corazawaf/coraza/v3/types"
+)
+
+type WAFConfig interface {
+	WithRule(rule *corazawaf.Rule) WAFConfig
+	WithRulesFromFile(path string) WAFConfig
+	WithRulesFromString(rules string) WAFConfig
+
+	WithAuditLog(config AuditLogConfig) WAFConfig
+
+	WithContentInjection() WAFConfig
+
+	WithRequestBodyAccess(config RequestBodyConfig) WAFConfig
+	WithResponseBodyAccess(config ResponseBodyConfig) WAFConfig
+
+	WithDebugLogger(logger corazawaf.DebugLogger) WAFConfig
+	WithErrorLogger(logger corazawaf.ErrorLogCallback) WAFConfig
+
+	WithFSRoot(fs fs.FS) WAFConfig
```

Comment:
I would expect something like `WithRootFS` or `WithRootDir` as it is more clear although the parameter is a Fs and not a Dir cc @jptosso